### ISSUE TITLE
Just a few suggestions

### ISSUE THREAD

**Issue by cfjello** [kisvegabor/abbreviations-in-code#11]
Great list! 

Here below you will find some additional suggestions:

chk  - 'check'
curr - the database name/prefix for 'current'
dyn - 'dynamic'
ext  - 'extension'
fac  - 'factory'
ins  -  'insert'
intf - 'interface' - I think this is more commonly used  than 'iface'
itor  -  'iterator'  (my own idea)
meta - 'Metadata' (my own idea)
mut - 'mutable' from Rust
ok   - 'success'
perf - 'performance'
rec  - 'record'
ref  -  'reference' like in object reference
regex - 'regular expression'
seq    - 'sequence'
tmpl - 'template'
tz - 'time zone'
upd - 'update'
util  - 'utility' like in utility function

**Comment by kisvegabor** [comment]
Hi,

Thank for the suggestions. Could you open a PR with them?

I'm not sure about `upd - 'update'`. I'm not sure I could figure out from "upd" that it stands for "update". I thin it should be added as "italic" meaning "context sensitive".

**Comment by philipeachille** [comment]
I see no issue with "upd"...

**Comment by kisvegabor** [comment]
Okay, it's democracy. If more people says it's ok, I'm ok with that too :slightly_smiling_face:

**Comment by cfjello** [comment]
'upd' is used in the database world

On closer inspection of your list I noticed a few things:

- req,  should be 'request' to match 'res' for 'response' (at least in a web-server context)
- You should remove/pass judgement on  the homographs, e.g.  v for both 'value' and 'vector' should be changed to 'val' for 'value' (as you have it) and 'vec' for 'vector' and 'v' should be removed or if kept it should stand for 'version'
- Where there are multiple possible meanings separated by '/' the default/preferred meaning should placed first, e.g. 'dir' should name 'directory' first.

If you do the above, I can provide a script to parse the README.md into various formats, JSON, JS/TS and even XML, format that can be generated and added to your github repo.

**Comment by philipeachille** [comment]
@cfjello I go with all your comments

**Comment by T1xx1** [comment]
Everywhere you want to abbreviate words and phrases: messages, emails, also in code in programs. If it would be machine readable I can try to build a VScode extension for that so all this abbreviations would be useful to someone. I'm also building an API for abbreviations and this repo would avoid me to rewrite everything again in the code.

**Comment by cfjello** [comment]
I agree with T1xx1. Naming conventions can be quite effective, e.g. If you are dealing with introspection and have an object instance that has a function with a ‘chk’ prefix, e.g. ‘chkRange()’ , you would know that it is a validation function you could call.

**Comment by kisvegabor** [comment]
Okay, got it. Thanks.

> if you use the above proposed "one line per meaning" you should stilll be able to indicate the preferred meaning (e.g. by adding the keyword 'default')

I'm ok with this approach and it'd really help to easily create machine readable versions. 

> Instead of strikethrough and italic

I also agree with it. Italic is hard to notice.  We could have a rule like:
```
- :optional-symbol: **abbreviation** long form (optional note)
```
For symbols I suggest:  :x: and :warning: 

What do yo think?

**Comment by T1xx1** [comment]
For me it's ok, but consider using 🔴🟡 instead. They are nicer and more readable.

**Comment by kisvegabor** [comment]
I'm fine with the circles too. :slightly_smiling_face:

**Comment by philipeachille** [comment]
i have another addition: "prod" for "production"

**Comment by kisvegabor** [comment]
My goal is just to have list of abbreviations as reference. I see the advantages of having the list in JSON (or other format) but it's outside of my personal interest. However if you send PRs I'd happily review and merge them.

For the suggest abbreviation a lot of things were said, and I think it'd be clearer to have a PR with them too.

**Comment by cfjello** [comment]
I have created a Pull Request with my changes + a script and a JSON file.

**Comment by kisvegabor** [comment]
I've merged #12.
